ext/date/php_date.c

```diff
@@ -1970,6 +1970,54 @@ static int implement_date_interface_handler(zend_class_entry *interface, zend_cl
 	return SUCCESS;
 } /* }}} */
 
+static int date_interval_has_property(zval *object, zval *member, int type, void **cache_slot) /* {{{ */
+{
+	php_interval_obj *obj;
+	zval tmp_member;
+	zval rv;
+	zval *prop;
+	int retval = 0;
+
+	if (Z_TYPE_P(member) != IS_STRING) {
+		ZVAL_COPY(&tmp_member, member);
+		convert_to_string(&tmp_member);
+		member = &tmp_member;
+		cache_slot = NULL;
```

Comment:
you could you zval_get_string like: https://github.com/php/php-src/blob/master/Zend/zend_object_handlers.c#L706